src/main/java/com/xceptance/neodymium/util/SelenideAddons.java

```diff
@@ -473,4 +474,120 @@ public static void openHtmlContentWithCurrentWebDriver(String htmlContent)
+     * @param waitingTime
+     *            the time to wait between retries
+     * @return if the element did match the condition within the given retries
+     */
+    public static boolean optionalWaitUntilCondition(SelenideElement element, Condition condition, long waitingTime)
+    {
+        boolean result = false;
+        int counter = 0;
+        while (counter < Neodymium.configuration().optionalElementRetryCount())
+        {
+            counter++;
+            if (element.has(condition))
+            {
+                result = true;
+                break;
+            }
+            Selenide.sleep(waitingTime);
+        }
+        return result;
+    }
+
+    /**
+     * Waits while an optional element matches a condition. This function will return false if the element does match
+     * the given condition or can not be found after the given timeframe. This method will use the default optional
+     * retry timeout.
+     * <p>
+     * The following settings can be configured within the Neodymium configuration to tune the retry behavior:
+     * </p>
+     * <ul>
+     * <li>*
+     * <li>neodymium.selenideAddons.optional.retry.timeout (default 2000ms pause between retries)</li>
+     * </ul>
+     * 
+     * @param element
+     *            the element to match
+     * @param condition
+     *            the condition for the element
+     * @return if the element did stop matching the condition within the given retries
+     */
+    public static boolean optionalWaitWhileCondition(SelenideElement element, Condition condition)
+    {
+        return optionalWaitWhileCondition(element, condition, Neodymium.configuration().optionalElementRetryTimeout());
+    }
+
+    /**
+     * Waits while an optional element matches a condition. This function will return false if the element does match
+     * the given condition or can not be found after the given timeframe.
+     * <p>
+     * The following settings can be configured within the Neodymium configuration to tune the retry behavior:
+     * </p>
+     * <ul>
+     * <li>neodymium.selenideAddons.optional.retry.count (default 5 retries)</li>
+     * </ul>
+     *
+     * @param element
+     *            the element to match
+     * @param condition
+     *            the condition for the element
+     * @param waitingTime
+     *            the time to wait between retries
+     * @return if the element did stop matching the condition within the given retries
+     */
+    public static boolean optionalWaitWhileCondition(SelenideElement element, Condition condition, long waitingTime)
+    {
+        boolean result = false;
+        int counter = 0;
+        while (counter < Neodymium.configuration().optionalElementRetryCount())
+        {
+            counter++;
+            if (element.has(not(condition)))
+            {
+                result = true;
+                break;
+            }
+            Selenide.sleep(waitingTime);
+        }
+        return result;
```

Comment:
I think, we can reuse the `optionalWaitUntilCondition` method here because `not(condition)` is also a condition
```suggestion
       return optionalWaitUntilCondition(element, not(condition), waitingTime);
```